### DISCUSSION TITLE
.NET Package for Windows ARM64

### DISCUSSION THREAD

**Discussion by linzeqipku** [google/or-tools#3677] [Feature requests]
**What language and solver does this apply to?**
C#

**Describe the problem you are trying to solve.**
Currently the nuget package [Google.OrTools](https://www.nuget.org/packages/Google.OrTools) supports `x86_64` for windows.
Therefore, my VS extension using `OrTools` cannot be used when the user platform is `windows arm64`.

**Describe the solution you'd like**
Do you have plan to also support `arm64` (for windows) in the near future? My target .NET platform is `net472`.

**Describe alternatives you've considered**
It would be helpful if you can share whether it is possible to build it from source code, and potential challenges in this work item.

**Additional context**

**Comment by lperron** [comment]
not in our plans. We do not have the hardware.

**Comment by fobrs** [comment]
I'm working on a port... scip needs a patch, but it's already working in my test case

**Comment by fobrs** [comment]
Here it is: [link](https://github.com/fobrs/or-tools/releases)

**Comment by Mizux** [comment]
dev note: Windows arm64 runners may be available to have a CI job...
https://github.blog/changelog/2025-04-14-windows-arm64-hosted-runners-now-available-in-public-preview/

**Reply by fobrs** [reply]
I know., they were already used to create an arm64 version of Git for Windows. But sadly I'm not enough educated in github to use or configure those runners, and I currently don't have time to figure out how to use them. However I can provide you with a windows batch file which builds both x64 and arm64 versions of OrTools for Windows dotnet (patched with my arm64 changes). It then uses the native protoc.exe to excecute when cross compiling.
